test/e2e/framework/pod/wait.go

```diff
@@ -226,13 +231,20 @@ func WaitForPodCondition(c clientset.Interface, ns, podName, desc string, timeou
 	return fmt.Errorf("Gave up after waiting %v for pod %q to be %q", timeout, podName, desc)
 }
 
+func podTimeout() time.Duration {
```

Comment:
For most of the e2e tests we run in prow the e2etest binary is compiled and runs in linux containers so I'm worried still return the linux timeout most of the time.

This should be checking if the pod is scheduled to a Windows nodes but I am having trouble recommending a generic way to do this in the e2e test framwork at the moment :(

Reply:
The best way that I can think of to accomplish this is to add a command-line argument to the e2etest binary to override the default pod timeout and then set that in the prow configs for the Windows test passes.

@adelina-t @claudiubelu can either of you think of a better way to set a global pod start-up timeout based on OS?

Reply:
My understanding is runtime.GOOS will check when the code is executed? I did similar change before and it is working?

Reply:
@marosset thanks for pointing it out. 
I changed to use framework.NodeOSDistroIs("windows"), I think this should work. When running the test nodeOSDistrols will set to "windows" for windows tests.

Reply:
Didn't know that existed but this should work nicely.